File: Coding/Perl/running_total.pl
sub running_total(){
    my (@param)=@{$_[0]};
    my $size=@param;
    my @sums;

    for($i=0;$i<$size;$i++){
        $sums[$i] = $param[$i] + $sums[$i-1];
    }
    return @sums;
}

my @elements = (1,2,3,4,5,6,7,8,9,10);#(5,15,23,68,91,12,35,46,81,3,12345,100,513,214,79,651,5,17,54,89);
@sums = &running_total(\@elements);
print "@sums\n";

File: Coding/C/running_total.c
#include<stdio.h>

int running_total(int [], int, int);

int main(){
	int elements[] = {1,2,3,4,5,6,7,8,9,10};//{5,15,23,68,91,12,35,46,81,3,12345,100,513,214,79,651,5,17,54,89};
	int n = sizeof(elements)/sizeof(int)-1;
	int sum = 0;
	sum = running_total(elements, n, sum);
	printf("%d is the final answer!\n",sum);
	return 0;
}

int running_total(int param[], int size, int sum){
	printf("%d is the running total.\n",sum);
	if(size>=0){
		sum += param[size];
	} else{
		return sum;
	}
	size--;
	return running_total(param, size, sum);
}